crates/core/tedge_api/src/mqtt_topics.rs

```diff
@@ -305,6 +305,75 @@ impl EntityTopicId {
         }
         .map(|parent_id| EntityTopicId(format!("device/{parent_id}//")))
     }
+
+    /// If `self` is a device topic id, return a service topic id under this
+    /// device.
+    ///
+    /// The device topic id must be in a format: "device/DEVICE_NAME//"; if not,
+    /// `None` will be returned.
+    pub fn to_service_topic_id(&self, service_name: &str) -> Option<ServiceTopicId> {
+        if let ["device", device_name, "", ""] = self.0.split('/').collect::<Vec<&str>>()[..] {
+            return Some(ServiceTopicId(EntityTopicId(format!(
+                "device/{device_name}/service/{service_name}"
+            ))));
+        }
+        None
+    }
+
+    /// Returns an array of all segments of this entity topic.
+    pub fn segments(&self) -> [&str; ENTITY_ID_SEGMENTS] {
+        let mut segments = self.0.split('/');
+        let seg1 = segments.next().unwrap();
+        let seg2 = segments.next().unwrap();
+        let seg3 = segments.next().unwrap();
+        let seg4 = segments.next().unwrap();
+        [seg1, seg2, seg3, seg4]
+    }
+
+    pub fn as_str(&self) -> &str {
+        self.0.as_str()
+    }
+}
+
+/// Represents an entity topic identifier known to be a service.
+///
+/// It's most often in a format `device/DEVICE_NAME/service/SERVICE_NAME`.
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+pub struct ServiceTopicId(EntityTopicId);
```

Comment:
Do we really need a type for `ServiceTopicId`? I don't think so: an entity topic id can be used for services as well as for devices.

Reply:
I think in some places (e.g. health checks) we do actually want to operate on services, and so to make sure a given topic is a service topic, we need to either look at it (if in default MQTT scheme) or call into entity store (if not). IMO it'd be nice not to write a guard every time to make sure that it is a service.
If I'm mistaken about there being more places where a function expects an entity topic id to be a service, then I can remove it for now and do it with plain `ServiceTopicId`s.